### ISSUE TITLE
Scripted inception_v3 has wrong output format

### ISSUE THREAD

**Issue by apivovarov** [pytorch/vision#6121]
### 🐛 Describe the bug

Scripted `inception_v3` model has wrong output format - Instead of `Tensor` it returns `InceptionOutputs`.
I checked other models - resnet50, mobilenet_v2 - After scripting they return `tensor`.
Inference tools such as torch_tensorrt expects models to return `Tensor` or `[Tensor]`.

To reproduce the issue with inception_v3:
```
import torch
import torchvision.models as models

model = models.inception_v3(pretrained=True).eval()
x = torch.rand(1, 3, 299, 299)
y=model(x)
smodel = torch.jit.script(model, x).eval()
y=smodel(x)

print(type(y))

```
It prints `<class 'torch._jit_internal.InceptionOutputs'>`.

Expected output: `<class 'torch.Tensor'>` 

What should be fixed in [inception.py](https://github.com/pytorch/vision/blob/main/torchvision/models/inception.py) code to make it output Tensor after applying `torch.jit.script`?

### Versions

```
PyTorch version: 1.12.0a0+bd13bc6
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.22.3
Libc version: glibc-2.31

Python version: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:10)  [GCC 10.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-1075-aws-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.6.124
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 515.43.04
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.4.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.4.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] pytorch-quantization==2.1.2
[pip3] torch==1.12.0a0+bd13bc6
[pip3] torch-tensorrt==1.1.0a0
[pip3] torchtext==0.13.0a0
[pip3] torchvision==0.13.0a0
[conda] magma-cuda110             2.5.2                         5    local
[conda] mkl                       2019.5                      281    conda-forge
[conda] mkl-include               2019.5                      281    conda-forge
[conda] numpy                     1.22.3           py38h1d589f8_2    conda-forge
[conda] pytorch-quantization      2.1.2                    pypi_0    pypi
[conda] torch                     1.12.0a0+bd13bc6          pypi_0    pypi
[conda] torch-tensorrt            1.1.0a0                  pypi_0    pypi
[conda] torchtext                 0.13.0a0                 pypi_0    pypi
[conda] torchvision               0.13.0a0                 pypi_0    pypi
```

**Comment by datumbox** [comment]
@apivovarov Thanks for reporting. First of all let me say that I agree with you that this is rather confusing/problematic.

InceptionV3 is a rather old implementation that uses a bunch of idioms which we no longer use in classification models. In this case, the model returns different output format conditionally on the training flag. The original idea of the author of this implementation was that the auxiliary parts of the model should be returned only when we are training and the Aux Logits are non-None. In that case, the model should return a tuple of the model output and the auxiliary out. So why JIT behaves differently? While Python is happy with the above dynamic return type, JIT isn't. So to circumvent that a tuple is created and returned always when JIT is used.

The above approach is only currently maintained to ensure BC. Perhaps this might be one of the things we can fix when we decide to revamp the Models area of TorchVision (cc @YosuaMichael). One approach could be to use one of the intermediate feature extraction utils we offer to extract the intermediate results from the main model and define a training-only variant with Auxiliary. At any case, note that the specific technique of adding auxiliary heads has also been phased out mostly from research, making the whole approach obsolete.
